metricflow/dataflow/sql_table.py

```diff
@@ -16,18 +16,13 @@ def _from_yaml_value(cls, input: PydanticParseableValueType) -> SqlTable:
         """Parses a SqlTable from string input found in a user-provided model specification
 
         Raises a ValueError on any non-string input, as all user-provided specifications of table identifiers
-        should be strings conforming to the expectations defined in the from_string method. The lone exception
-        is for SqlTable instance inputs, which could arise if we are initializing Materialization objects internally
-        without direct access to user input.
+        should be strings conforming to the expectations defined in the from_string method.
         """
         if isinstance(input, str):
             return SqlTable.from_string(input)
-        elif isinstance(input, SqlTable):
-            # This is internally constructed, pass it through and ignore it in error messaging
-            return input
```

Comment:
This was added in 15352c70f641b3e09680b790fe51a47399d6ff81. It looks like we added it for cases where we internally build a model. Do we expect to not need to internally build the model or is there different reasoning?

Reply:
To answer my own question, [metricflow/model/objects/base.py#147](https://github.com/transform-data/metricflow/blob/4536401243884fc45cd970b0b9395b10444d96b0/metricflow/model/objects/base.py#L147) handles the case wherein the input is an instance of the desired class itself.